### PR TITLE
fixes the speed of vapes and makes them use delta_time

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

The creator of https://github.com/tgstation/tgstation/pull/59578 misinterpreted drag_cooldown as being the cooldown for all vape reagent transfers instead of the cooldown for just the big smoke clouds made by super and emagged vapes. As a result, all vapes had their effective reagent transfer rates cut from 0.2u every ~second to 0.2u every ~EIGHT seconds.

This PR fixes that error by moving handle_reagents() to above the cooldown check and renames drag_cooldown to super_drag_cooldown so that it won't be repeated.

Also, I made the vape version of handle_reagents() account for delta_time.

None of these changes affect cigarettes.

## Why It's Good For The Game

This (significant!) change was not mentioned anywhere in the PR body, changelog, or comments of https://github.com/tgstation/tgstation/pull/59578, so I can only assume that it was unintentional. If making topped up vapes last for over an hour each WAS intentional, feel free to correct me in the comments, Oroboros.

I could theoretically make the chance for an emagged vape to explode when puffed use delta_time, but it's already behind an 8 second cooldown anyway and I don't want to accidentally bork that probability due to a misunderstanding of how DT_PROB() works.

## Changelog
:cl: ATHATH
fix: An unintentional change to the frequency at which vapes transferred their reagents at has been fixed, and full vapes should no longer last for over an hour each.
code: Vapes now account for delta_time in their reagent transfers, making them more consistent.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
dunno what's causing the errors here

**Comment by CRITAWAKETS** [comment]
You accidentally changed drag_cooldown to super_drag_cooldown without changing the name of the var or adding a new one called that.

**Comment by TemporalOroboros** [comment]
Honestly didn't notice this change. I think I just autopiloted through making vapes use the same logic as cigarettes. Thanks for catching this.

**Comment by AMonkeyThatCodes** [comment]
adding delta_time to something without changing the used values is a balance change

**Comment by ATH1909** [comment]
> adding delta_time to something without changing the used values is a balance change

?

**Comment by AMonkeyThatCodes** [comment]
deltatime is 2 for most subsystems. if you add * deltatime you're just doubling every value in the calculation.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Floyd is right, these are on SSobj which have a delta time of 2. This would change the behavior. Everywhere you are currently using delta_time, divide it by two.

**Comment by ATH1909** [comment]
> Floyd is right, these are on SSobj which have a delta time of 2. This would change the behavior. Everywhere you are currently using delta_time, divide it by two.

... huh. Okay, then.

**Comment by ATH1909** [comment]
> Floyd is right, these are on SSobj which have a delta time of 2. This would change the behavior. Everywhere you are currently using delta_time, divide it by two.

done

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -1025,28 +1025,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 	var/mob/living/carbon/vaper = loc
 	if(!iscarbon(vaper) || src != vaper.wear_mask)
-		reagents.remove_any(REAGENTS_METABOLISM)
+		reagents.remove_any(REAGENTS_METABOLISM * delta_time * 0.5)
 		return
 
 	if(reagents.get_reagent_amount(/datum/reagent/fuel))
 		//HOT STUFF
-		vaper.adjust_fire_stacks(2)
+		vaper.adjust_fire_stacks(2 * delta_time * 0.5)
```
Comment (AMonkeyThatCodes) [inline code/game/objects/items/cigs_lighters.dm:1039]: bruh just do 1 * delta_time

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -1025,28 +1025,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 		var/datum/effect_system/reagents_explosion/e = new()
-		e.set_up(round(reagents.get_reagent_amount(/datum/reagent/toxin/plasma) / 2.5, 1), get_turf(src), 0, 0)
+		e.set_up(round(reagents.get_reagent_amount(/datum/reagent/toxin/plasma) / 2.5, 1), get_turf(src), 0, 0) //this is an explosion that destroys the vape, so it should NOT be affected by delta_time
 		e.start()
 		qdel(src)
 
-	if(!reagents.trans_to(vaper, REAGENTS_METABOLISM, methods = INGEST, ignore_stomach = TRUE))
-		reagents.remove_any(REAGENTS_METABOLISM)
+	if(!reagents.trans_to(vaper, REAGENTS_METABOLISM * delta_time * 0.5, methods = INGEST, ignore_stomach = TRUE))
+		reagents.remove_any(REAGENTS_METABOLISM * delta_time * 0.5)
```
Comment (AMonkeyThatCodes) [inline code/game/objects/items/cigs_lighters.dm:1049]: just make these variables or something

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (ATH1909) [reply code/game/objects/items/cigs_lighters.dm:1039]: it's to make it clear the delta_time is supposed to be 2-ish for this proc

File: code/game/objects/items/cigs_lighters.dm (hunk 2 above)
Comment (ATH1909) [reply code/game/objects/items/cigs_lighters.dm:1049]: eh, REAGENTS_METABOLISM * delta_time * 0.5 is basically just used here, and no vape subtypes exist yet

Comment (AMonkeyThatCodes) [reply code/game/objects/items/cigs_lighters.dm:1049]: scalability exists for the future, not just for current convenience.

**Pull request closed without merging**